js/hooks/useNfc.ts

```diff
@@ -0,0 +1,66 @@
+import { nfcSupported } from "../util/nfc";
+import { useEffect, useState } from "react";
+
+export type NfcResult =
+  | { status: "nfcUnsupported" }
+  | { status: "reading" }
+  | { status: "success"; data: string }
+  | { status: "cancelled" }
+  | { status: "error"; error: Error };
+
+export const useNfc = (cancel: AbortController): NfcResult => {
+  const supported = nfcSupported();
+
+  const [result, setResult] = useState<NfcResult>({
+    status: supported ? "reading" : "nfcUnsupported",
+  });
+
+  useEffect(() => {
+    if (supported) {
+      const reader = new NDEFReader();
+
+      reader.scan({ signal: cancel.signal }).catch((error: unknown) => {
+        if (error instanceof Error) {
```

Comment:
This would work fine if it was a String still, right?

Reply:
Yeah, thinking about it it doesn't really have to be any type in particular. See 18cd5b2.